prov/efa/src/rxr/rxr_env.c

```diff
@@ -42,6 +42,7 @@ struct rxr_env rxr_env = {
 	.tx_min_credits = RXR_DEF_MIN_TX_CREDITS,
 	.tx_queue_size = 0,
 	.enable_shm_transfer = 1,
+	.enable_eager_message = 1,
```

Comment:
maybe just `enable_eager` or `efa_eager_protocol`?

Reply:
Why have an enable at all?  Just have a max_eager_size.  0 == no eager protocol.  Much more flexible, for no real cost.

Reply:
`max_eager_size` is not a constant though. It depends on packet size and header size.

Reply:
@bwbarrett I think that runs into an issue with the medium message protocol. I'm assuming when you say no eager, you don't want that executing either.

Reply:
So make `max_eager_size` a run-time constant.  You know the packet size fairly early in initializing the EFA device.  THe difference between the max and min header size is a small fraction of expected packet sizes.  Save yourself the complexity (and computation time!) and just figure out the max header size during initialization, then use that as the eager limit.  Don't over-optimize for a couple bytes on the eager / medium threshold.

Reply:
For the medium / runting (note: I don't think you need a medium AND a runting protocol.  The runting protocol really is what the medium protocol always should have been (assuming the data in flight limit is sane).  But in that case, you'd set a threshold for the max runting / max medium threshold, and to avoid copies entirely, you'd set both the eager and runting / medium message to 0.

Reply:
I agree with Brian, to avoid copies entirely, you need to set both the eager and runting / medium message to 0.